ambry-router/src/test/java/com.github.ambry.router/GetBlobInfoOperationTest.java

```diff
@@ -453,6 +513,45 @@ private void testVariousErrors(String dcWherePutHappened) throws Exception {
     assertSuccess(op);
   }
 
+  /**
+   * Assert that operation fails with the expected error code
+   * @param errorCode expected error code on failure
+   * @throws RouterException
+   */
+  private void assertOperationFailure(RouterErrorCode errorCode)
+      throws RouterException, IOException, InterruptedException {
+    NonBlockingRouter.currentOperationsCount.incrementAndGet();
+    GetBlobInfoOperation op =
+        new GetBlobInfoOperation(routerConfig, routerMetrics, mockClusterMap, responseHandler, blobIdStr, options, null,
+            routerCallback, kms, cryptoService, cryptoJobHandler, time);
+    ArrayList<RequestInfo> requestListToFill = new ArrayList<>();
+    requestRegistrationCallback.requestListToFill = requestListToFill;
+    op.poll(requestRegistrationCallback);
+    Assert.assertEquals("There should only be as many requests at this point as requestParallelism", requestParallelism,
+        correlationIdToGetOperation.size());
+
+    List<ResponseInfo> responses = sendAndWaitForResponses(requestListToFill);
+    for (ResponseInfo responseInfo : responses) {
+      GetResponse getResponse = responseInfo.getError() == null ? GetResponse.readFrom(
+          new DataInputStream(new ByteBufferInputStream(responseInfo.getResponse())), mockClusterMap) : null;
+      op.handleResponse(responseInfo, getResponse);
+      if (op.isOperationComplete()) {
+        break;
+      }
+    }
+
+    int timeout = 500;
```

Comment:
this loop could potentially be moved into a helper method since it seems to be used in a couple places. Do you know how many iterations this usually runs for? I can see that this might be hard to do in a deterministic way